src/Sulu/Bundle/WebsiteBundle/Cache/CacheClearer.php

```diff
@@ -69,17 +69,28 @@ public function __construct(
         $this->eventDispatcher = $eventDispatcher;
     }
 
-    public function clear()
+    public function clear(/*?array $tags = []*/)
     {
+        $tags = \func_num_args() >= 1 ? \func_get_arg(0) : null;
+
         if ($this->cacheManager && $this->cacheManager->supportsInvalidate()) {
+            if (!empty($tags) && $this->cacheManager->supportsInvalidateTag()) {
+                foreach ($tags as $tag) {
+                    $this->cacheManager->invalidateTag($tag);
+                }
+
+                $this->eventDispatcher->dispatch(new CacheClearEvent($tags), Events::CACHE_CLEAR);
+
+                return;
```

Comment:
would move the code below into a `else` block. that way its easier to understand that we either call `invalidateTag` or `invalidateDomain`

Reply:
Not sure with an else here I think else block make it less readable then the early return we mostly did go with early returns instead of else statements.

Reply:
dont have a too strong opinion on this, we can leave it like this too 🙂

Reply:
i am of the same opinion as @alexander-schranz - will keep it